### PR TITLE
use new Timestamp API

### PR DESCRIPTION
**Pull request by colinsurprenant**

### CODE REVIEW

File: lib/logstash/outputs/mongodb.rb [hunk 1]
```diff
@@ -56,9 +55,9 @@ def receive(event)
...
         # the mongodb driver wants time values as a ruby Time object.
         # set the @timestamp value of the document to a ruby Time object, then.
-        document = event.to_hash
+        document = event.to_hash.merge("@timestamp" => event.timestamp.time)
       else
-        document = event.to_hash.merge("@timestamp" => event["@timestamp"].to_json)
+        document = event.to_hash.merge("@timestamp" => event.timestamp.to_iso8601)
       end
       if @generateId
         document['_id'] = BSON::ObjectId.new(nil, event["@timestamp"])
```
Comment (jsvd) [inline lib/logstash/outputs/mongodb.rb:63]: why not here as well?

File: lib/logstash/outputs/mongodb.rb [hunk 2]
```diff
@@ -56,9 +55,9 @@ def receive(event)
       if @isodate
         # the mongodb driver wants time values as a ruby Time object.
         # set the @timestamp value of the document to a ruby Time object, then.
-        document = event.to_hash
+        document = event.to_hash.merge("@timestamp" => event.timestamp.time)
```
Comment (jsvd) [inline lib/logstash/outputs/mongodb.rb:58]: this changes current behavior, is it expected?

**Comment by purbon** [comment]
@colinsurprenant @jsvd any reason not to merge this? can we move forward and get this in?

**Comment by guyboertje** [comment]
@colinsurprenant, @purbon, @jsvd - I am closing this in favour of adding a BSON patch file that patches any objects that are missed by [BSON](https://github.com/mongodb/bson-ruby/tree/master/lib/bson).

Lets kill these to_bson and bson_type method missing errors for once and for all. Refer to [this comment](https://github.com/logstash-plugins/logstash-output-mongodb/issues/9#issuecomment-160574867)

**Comment by purbon** [comment]
Makes complete sense to me.

/purbon

**Pull request closed without merging**